generator/src/test/java/com/scottlogic/deg/generator/cucumber/LessThan.feature

```diff
@@ -2,12 +2,332 @@ Feature: User can specify that a numeric value is lower than, but not equal to,
+     Given foo is less than 2
+       And foo is anything but less than 2
+     Then no data is created
+
+#lessThanOrEqual
+@ignore #589: produces big-integer numbers rather than integer values, not sure why
+Scenario: lessThan run against a non contradicting lessThanOrEqualTo should be successful (lessThan 6 AND lessThanOrEqualTo 5)
+     Given foo is less than 6
+       And foo is less than or equal to 5
+       And the generator can generate at most 5 rows
+     Then the following data should be generated:
+       | foo         |
+       | -2147482624 |
+       | -2147482623 |
+       | -2147482622 |
+       | -2147482621 |
+       | -2147482620 |
+
+Scenario: lessThan run against a non contradicting not lessThanOrEqualTo should be successful (lessThan 10 AND not lessThanOrEqualTo 2)
+     Given foo is less than 10
+       And foo is anything but less than or equal to 2
+     Then the following data should be generated:
+       | foo      |
+       | 3        |
+       | 4        |
+       | 5        |
+       | 6        |
+       | 7        |
+       | 8        |
+       | 9        |
+
+Scenario: not lessThan run against a non contradicting lessThanOrEqualTo should be successful (not lessThan 2 AND lessThanOrEqualTo 10)
+     Given foo is anything but less than 2
+       And foo is less than or equal to 10
+     Then the following data should be generated:
+       | foo      |
+       | 2        |
+       | 3        |
+       | 4        |
+       | 5        |
+       | 6        |
+       | 7        |
+       | 8        |
+       | 9        |
+       | 10       |
+
+Scenario: not lessThan run against a non contradicting not lessThanOrEqualTo should be successful (not lessThan 3 AND not lessThanOrEqualTo 4)
+     Given foo is anything but less than 3
+       And foo is anything but less than or equal to 4
+       And the generator can generate at most 5 rows
+     Then the following data should be generated:
+       | foo |
+       | 5   |
+       | 6   |
+       | 7   |
+       | 8   |
+       | 9   |
+
+@ignore #589: produces big-integer numbers rather than integer values, not sure why
+Scenario: lessThan run against a contradicting lessThanOrEqualTo should only only generate string, temporal and null (lessThan 1 AND lessThanOrEqualTo 1)
+     Given foo is less than 1
+       And foo is less than or equal to 1
+       And the generator can generate at most 5 rows
+     Then the following data should be generated:
+       | foo         |
+       | -2147482624 |
+       | -2147482623 |
+       | -2147482622 |
+       | -2147482621 |
+       | -2147482620 |
+
+Scenario: not lessThan run against a contradicting not lessThanOrEqualTo should only only generate string, temporal and null (lessThan 2 AND not lessThanOrEqualTo 3)
+     Given foo is less than 2
+       And foo is anything but less than or equal to 3
+     Then no data is created
```

Comment:
Remove anything but null from the background so that soft contradictions can be validated against the creation of null data.

Reply:
Would this add value to the test or just assert that null can be emitted when numeric constraints contradict?